mercantile/__init__.py

```diff
@@ -341,45 +338,140 @@ def parent(*tile):
-    # Algorithm ported directly from https://github.com/mapbox/tilebelt.
-    if xtile % 2 == 0 and ytile % 2 == 0:
-        return Tile(xtile // 2, ytile // 2, zoom - 1)
-    elif xtile % 2 == 0:
-        return Tile(xtile // 2, (ytile - 1) // 2, zoom - 1)
-    elif not xtile % 2 == 0 and ytile % 2 == 0:
-        return Tile((xtile - 1) // 2, ytile // 2, zoom - 1)
-    else:
-        return Tile((xtile - 1) // 2, (ytile - 1) // 2, zoom - 1)
-
-
-def children(*tile):
-    """Get the four children of a tile
+    if len(tile) == 2:
+        raise ValueError(
+            "Could not parse tile! Make sure that if you are calling this with zoom, you call this with zoom as a keyword argument."
+        )
+
+    if zoom is not None and (tile[2] < zoom or zoom != int(zoom)):
+        raise ValueError(
+            "zoom must be an integer and less than the source tile!")
+
+    x, y, z = tile
+    if x != int(x) or y != int(y) or z != int(z):
+        raise ValueError("Cannot find the parent of a fractional tile!")
+    target_zoom = z - 1 if zoom is None else zoom
+
+    # Algorithm heavily inspired by https://github.com/mapbox/tilebelt.
+    return_tile = tile
+    while return_tile[2] > target_zoom:
+        xtile, ytile, ztile = return_tile
+        if xtile % 2 == 0 and ytile % 2 == 0:
+            return_tile = Tile(xtile // 2, ytile // 2, ztile - 1)
+        elif xtile % 2 == 0:
+            return_tile = Tile(xtile // 2, (ytile - 1) // 2, ztile - 1)
+        elif not xtile % 2 == 0 and ytile % 2 == 0:
+            return_tile = Tile((xtile - 1) // 2, ytile // 2, ztile - 1)
+        else:
+            return_tile = Tile((xtile - 1) // 2, (ytile - 1) // 2, ztile - 1)
+    return return_tile
+
+
+def children(*tile, **kwargs):
+    """Get the children of a tile
 
     Parameters
     ----------
     tile : Tile or sequence of int
         May be be either an instance of Tile or 3 ints, X, Y, Z.
+    zoom : int, optional
+        Returns all children at zoom *zoom*, in depth-first clockwise winding order.
+        If unspecified, returns the immediate (i.e. zoom + 1) children of the tile.
 
     Returns
     -------
     list
     """
+    zoom = kwargs.get("zoom", None)
+
     if len(tile) == 1:
         tile = tile[0]
-    xtile, ytile, zoom = tile
-    return [
-        Tile(xtile * 2, ytile * 2, zoom + 1),
-        Tile(xtile * 2 + 1, ytile * 2, zoom + 1),
-        Tile(xtile * 2 + 1, ytile * 2 + 1, zoom + 1),
-        Tile(xtile * 2, ytile * 2 + 1, zoom + 1)]
+    if len(tile) == 2:
+        raise ValueError(
+            "Could not parse tile! Make sure that if you are calling this with zoom, you call this with zoom as a keyword argument."
+        )
+    xtile, ytile, ztile = tile
+
+    if zoom is not None and (ztile > zoom or zoom != int(zoom)):
+        raise ValueError(
+            "zoom must be an integer and greater than the source tile!")
+
+    target_zoom = zoom if zoom is not None else ztile + 1
+
+    queue = [tile]
```

Comment:
I feel like queue is not quite the right name.

``suggestion
    tiles = [tile]
```